core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -2022,12 +1947,7 @@ private[spark] object JsonProtocolSuite extends Assertions {
       |      "Port": 300
       |    },
       |    "Block ID": "rdd_0_0",
-      |    "Storage Level": {
```

Comment:
We'll also want to retain some tests of the old format to ensure it's still read. Maybe there are outside of the diff and I'm not seeing them.

Reply:
Added a to/from JSON test for a custom `StorageLevel`.

Reply:
I guess I mean, doesn't this no longer test whether it can read the verbose, old style format? like this test does here and the ones above, that are being removed?

Reply:
These strings are used with `testEvent` which assert that the serialized representation matches the one given in the string literal. See https://github.com/criteo-forks/spark/blob/7869e63a569a6fb6725996084f0c5c55fc130ac8/core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala#L457

Reply:
I share sean's concern, I don't think your response is addressing it.  You added one test on L151 makes sure that some event which is *not* predefined still works.  But you don't have a test making sure that the old, verbose string can still be parsed (or is it somewhere else?)

probably this is indirectly covered by HistoryServerSuite but a more direrct test would be better

Reply:
I've added a test ensuring all predefine storage levels can be read from the legacy format.

Sidenote: I've also noticed that the legacy format incorrectly handled the predefined `StorageLevel.OFF_HEAP` and an fact any other custom storage level with `useOffHeap = true`. It looks like a bug to me, wdyt?

Reply:
yup, I completely agree that off heap is not respected in the json format.  can you file a bug?  I think its still relevant even after this goes in, for custom levels